include/cutlass/gemm/device/gemm.h

```diff
@@ -362,7 +362,18 @@ class Gemm {
     if (!kSplitKSerial && args.split_k_slices > 1) {
       return Status::kErrorInvalidProblem;
     }
+    
+    // MaxDynamicSharedMemorySize must be smaller than the device attribute cudaDevAttrMaxSharedMemoryPerBlockOptin minus the function attribute sharedSizeBytes  
+    int smem_size = int(sizeof(typename GemmKernel::SharedStorage));
+    cudaFuncAttributes attr;
+    cudaFuncGetAttributes(&attr, Kernel<GemmKernel>);
+    int max_mem = 0;
+    cudaDeviceGetAttribute(&max_mem, cudaDevAttrMaxSharedMemoryPerBlockOptin, 0);
```

Comment:
we can change the logic here that only checking when `smem_size` is larger than 48KB.  then we dont need to get `attr`.

Reply:
Instead, we can also use 

```
        cudaDeviceProp properties;
        result = cudaGetDeviceProperties(&properties, device_idx);

        if (result != cudaSuccess) {
          return -1;
        }

        smem_capacity = static_cast<int>(properties.sharedMemPerMultiprocessor);
```

Reply:
> we can change the logic here that only checking when `smem_size` is larger than 48KB. then we dont need to get `attr`.
Wouldn't this limit the capacity of newer architectures with more shared memory available?

Reply:
Is the `sharedMemPerMultiprocessor` property equivalent to the original query? I can't infer this from the documentation

Reply:
> Wouldn't this limit the capacity of newer architectures with more shared memory available?

48KB is the limit of static allocated memory.  No plan to change.

> Is the sharedMemPerMultiprocessor property equivalent to the original query?

Feel free to try it first.